[Unified PDF] some text labels grow/shrink slightly when selecting radio buttons and clicking form-reset buttons

hortont424 opened the pull request:
#### 7d9cf59af5c4e17a3c8803a7b22a26cf1c90eacd
<pre>
[Unified PDF] some text labels grow/shrink slightly when selecting radio buttons and clicking form-reset buttons
<a href="https://bugs.webkit.org/show_bug.cgi?id=277971">https://bugs.webkit.org/show_bug.cgi?id=277971</a>
<a href="https://rdar.apple.com/132020944">rdar://132020944</a>

Reviewed by Simon Fraser.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/AsyncPDFRenderer.mm:
(WebKit::AsyncPDFRenderer::enqueueTilePaintIfNecessary):
Enclose repaint clip rects before making ImageBuffers with them, to avoid
fractional squishing when copying the ImageBuffers into the tiles.

Canonical link: <a href="https://commits.webkit.org/282135@main">https://commits.webkit.org/282135@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f535775821291852495fc615a87152b4877c07d5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62214 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41569 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14806 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/12759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49255 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13099 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/66194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/12759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65283 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/38610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/53882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/35296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/11690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/57010 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11460 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67924 "Failed to checkout and rebase branch from PR 32056") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6157 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/11219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/67924 "Failed to checkout and rebase branch from PR 32056") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6184 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53865 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/67924 "Failed to checkout and rebase branch from PR 32056") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13822 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5084 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/37368 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/38452 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/39548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/38197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/AsyncPDFRenderer.mm:
```diff
@@ -370,7 +370,13 @@
-    auto renderInfo = renderInfoForTile(tiledBacking, tileInfo, tileRect, clipRect);
+    // Round the clip rect to integer bounds so that we don't end up making
+    // ImageBuffers with floating point sizes.
+    std::optional<FloatRect> enclosingClipRect;
+    if (clipRect)
+        enclosingClipRect = enclosingIntRect(*clipRect);
+
+    auto renderInfo = renderInfoForTile(tiledBacking, tileInfo, tileRect, enclosingClipRect);
```
[C++23](https://en.cppreference.com/w/cpp/utility/optional/transform)!

```suggestion
    auto renderInfo = renderInfoForTile(tiledBacking, tileInfo, tileRect, clipRect.transform([](const auto& rect) {
        return enclosingIntRect(rect);
    }));
```

hortont424 replied:
!!

hortont424 committed:
[Unified PDF] some text labels grow/shrink slightly when selecting radio buttons and clicking form-reset buttons
https://bugs.webkit.org/show_bug.cgi?id=277971
rdar://132020944

Reviewed by Simon Fraser.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/AsyncPDFRenderer.mm:
(WebKit::AsyncPDFRenderer::enqueueTilePaintIfNecessary):
Enclose repaint clip rects before making ImageBuffers with them, to avoid
fractional squishing when copying the ImageBuffers into the tiles.

Canonical link: https://commits.webkit.org/282135@main
